frontend/pages/api/v1/cmc/supply.ts

```diff
@@ -0,0 +1,143 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { PythBalance } from '@pythnetwork/staking/app/pythBalance'
+import BN from 'bn.js'
+import { STAKING_ADDRESS } from '@pythnetwork/staking/app/constants'
+import { PYTH_TOKEN } from '@pythnetwork/staking/app/deploy/mainnet_beta'
+import { Connection, Keypair } from '@solana/web3.js'
+import { Program, AnchorProvider, IdlAccounts } from '@coral-xyz/anchor'
+import NodeWallet from '@coral-xyz/anchor/dist/cjs/nodewallet'
+import { Staking } from '@pythnetwork/staking/lib/target/types/staking'
+import idl from '@pythnetwork/staking/target/idl/staking.json'
+import { splTokenProgram } from '@coral-xyz/spl-token'
+import { TOKEN_PROGRAM_ID } from '@solana/spl-token'
+import { getConfig } from './../locked_accounts'
+import {
+  getAllCustodyAccounts,
+  getAllMetadataAccounts,
+  getAllStakeAccounts,
+} from '../all_locked_accounts'
+
+const connection = new Connection(process.env.BACKEND_ENDPOINT!)
+const provider = new AnchorProvider(
+  connection,
+  new NodeWallet(new Keypair()),
+  {}
+)
+const stakingProgram = new Program<Staking>(
+  idl as Staking,
+  STAKING_ADDRESS,
+  provider
+)
+const tokenProgram = splTokenProgram({
+  programId: TOKEN_PROGRAM_ID,
+  provider: provider as any,
+})
+
+export default async function handlerSupply(
```

Comment:
does this implement some CMC-defined API? It seems like this either needs some documentation or a link to the CMC API docs.